src/mono/mono/metadata/sre.c

```diff
@@ -3665,6 +3665,8 @@ typebuilder_setup_one_field (MonoDynamicImage *dynamic_image, MonoClass *klass,
 			field->type = mono_reflection_type_get_handle ((MonoReflectionType*)fb->type, error);
 			goto_if_nok (error, leave);
 		}
+		if (klass->enumtype && strcmp (field->name, "value__") == 0) // used by enum classes to store the instance value
+			field->type->attrs |= FIELD_ATTRIBUTE_RT_SPECIAL_NAME;
 
 		if (!klass->enumtype && !mono_type_get_underlying_type (field->type)) {
```

Comment:
Unrelated, but this look like a typo (!klass->enumtype?)

Reply:
yeah looks like a typo indeed. the check was added by https://github.com/mono/mono/commit/c21ffa3695045322136c50c7918633958c1113b8.

can you please open an issue about it so we don't forget?